lib/readmore.dart

```diff
@@ -174,26 +238,38 @@ class ReadMoreTextState extends State<ReadMoreText> {
 
   Widget _builder(BuildContext context, bool isCollapsed, Widget? child) {
     final defaultTextStyle = DefaultTextStyle.of(context);
-    TextStyle? effectiveTextStyle = widget.style;
-    if (widget.style?.inherit ?? false) {
+    var effectiveTextStyle = widget.style;
+    if (widget.style == null || widget.style!.inherit) {
       effectiveTextStyle = defaultTextStyle.style.merge(widget.style);
-    } else {
```

Comment:
It is a regression of https://github.com/jonataslaw/readmore/pull/35 ?
I couldn't test to be sure if this brings back the problem, but we need to double check this.

Reply:
The current solution is equivalent to using [`Text` / `Text.rich`](https://github.com/flutter/flutter/blob/d12ba5c270d83c63cd3c1e89c1cd1f279bbb5696/packages/flutter/lib/src/widgets/text.dart#L640).

This will only be false if there is an explicit `TextStyle` given with `inherit: false`, which is `true` by default.
```dart
var effectiveTextStyle = widget.style;
if (widget.style == null || widget.style!.inherit) {
  effectiveTextStyle = defaultTextStyle.style.merge(widget.style);
}
```

This means that if the programmer does not want to inherit the `DefaultTextStyle`, creating an empty `TextStyle` is acceptable in this case.

Reply:
```dart
final defaultTextStyle = DefaultTextStyle.of(context);
var effectiveTextStyle = widget.style;
if (widget.style == null || widget.style!.inherit) {
  effectiveTextStyle = defaultTextStyle.style.merge(widget.style);
}
```

The `effectiveTextStyle` will never be null:
- If `widget.style` is null, then `effectiveTextStyle` will be equal to `defaultTextStyle`.
- If `widget.style` has `inherit` set to true, `effectiveTextStyle` will be a merge of `defaultTextStyle` and `widget.style`.
- If `widget.style` has `inherit` set to false, `effectiveTextStyle` will be set according to `widget.style` and will not be overwritten by `defaultTextStyle`.

But I rewrote logic to make it more explicit:

```dart
final defaultTextStyle = DefaultTextStyle.of(context);
TextStyle effectiveTextStyle;
if (widget.style == null || widget.style!.inherit) {
  effectiveTextStyle = defaultTextStyle.style.merge(widget.style);
} else {
  effectiveTextStyle = widget.style!;
}
```